src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php

```diff
@@ -36,6 +36,8 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
             $prototype = $builder->create($options['prototype_name'], $options['entry_type'], $prototypeOptions);
             $builder->setAttribute('prototype', $prototype->getForm());
+
+            $resizeFormListenerPrototypeOptions = array_replace($options['entry_options'], $options['prototype_options']);
```

Comment:
What about the options set on L29-30, do we want them in this array?
Sorry I'm not much into this domain so this might be a dumb question.

Reply:
Very good question, I was wondering about that as well.

Those provide some default values in case they aren't provided via the `entry_options` and `prototype_options` array. Since required defaults to true, an collection with required set to false would have the entries set to required. (See issue #15544)

However since the `entry_options` and later the `prototype_options` have been introduced, this behavior is kind of redundant.

Same for the label, this is because otherwise the label isn't rendered as a replaceable string when using `prototype_name` as a label. Since the prototype is normally used to add new fields to a form using JS, it's important to have them be replaceable (see issue #3738). For the new fields rendered server-side, the `prototype_name` isn't used, so the label for the prototype shouldn't be used.
I'm not sure of the use case for falling back to the `prototype_name` as a label, but it's being kept the same for BC reasons.

So I think we don't want those values in the array and only use `entry_options` and `prototype_options` for the `ResizeFormListener`.